pxtblocks/fields/field_melodySandbox.ts

```diff
@@ -0,0 +1,415 @@
+
+            let gridDiv = goog.dom.createDom("div", {}) as HTMLElement;
+            gridDiv.className = "melody-grid-div";
+            for (let i = 0; i < this.numRow; i++) {
+                let row = document.createElement("div");
+                row.className = "row" + i;
+                for (let j = 0; j < this.numCol; j++) {
+                    let cell = document.createElement("button"); // button
+                    cell.className = "cell";
+                    cell.id = "cell-" + i + "-" + j;
+                    row.appendChild(cell);
+                }
+                gridDiv.appendChild(row);
+            }
+            div.appendChild(gridDiv);
+
+            let bottomDiv = goog.dom.createDom("div", {}) as HTMLElement;
+            bottomDiv.className = "melody-bottom-bar-div";
+
+            let buttonBarDiv = goog.dom.createDom("div", {}) as HTMLElement;
+            buttonBarDiv.className = "melody-button-bar-div";
+
+            let doneButton = document.createElement("button");
+            doneButton.id = "melody-done-button";
+            doneButton.className = "ui button";
+            doneButton.innerText = "Done";
+            doneButton.addEventListener("click", () => Blockly.DropDownDiv.hideIfOwner(this));
+            let iconButtons = goog.dom.createDom("div", {}) as HTMLElement;
+            iconButtons.className = "ui icon buttons";
+            let playButton = document.createElement("button");
+            playButton.className = "ui button";
+            playButton.id = "melody-play-button";
+            playButton.addEventListener("click", () => this.playMelody());
+            let playIcon = document.createElement("i");
+            playIcon.id = "melody-play-icon";
+            playIcon.className = "play icon";
+            playButton.appendChild(playIcon);
+            let loopButton = document.createElement("button");
+            loopButton.className = "ui button";
+            loopButton.id = "melody-loop-button";
+            loopButton.addEventListener("click", () => this.loopMelody());
+            let loopIcon = document.createElement("i");
+            loopIcon.className = "redo xicon";
+            loopButton.appendChild(loopIcon);
+
+            iconButtons.appendChild(playButton);
+            iconButtons.appendChild(loopButton);
+            buttonBarDiv.appendChild(iconButtons);
+            buttonBarDiv.appendChild(doneButton);
+
+            let sliderDiv = goog.dom.createDom("div", {}) as HTMLElement;
+            sliderDiv.className = "slider-container";
+
+            let tempoSlider = document.createElement("input");
+            tempoSlider.id = "melody-tempo-slider";
+            tempoSlider.type = "range";
+            tempoSlider.className = "ui range";
+            tempoSlider.min = "60";
+            tempoSlider.max = "200";
+            tempoSlider.defaultValue = this.tempo + "";
+            tempoSlider.addEventListener("input", () => this.setTempo(+tempoSlider.value));
```

Comment:
Not sure what the + is for. Same below

Reply:
It's for converting the string to a number.